### ISSUE TITLE
Cannot inject providers into useFactory (async) - incompatible "unknown"

### ISSUE THREAD

**Issue by appsolutegeek** [nestjs/terminus#158]
<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
I am trying to use "useFactory" of forRootAsync

like so 

```
   TerminusModule.forRootAsync({
      imports: [HealthModule],
      inject: [ConfigService, TerminusOptionsService],
      useFactory: async (
        configService: ConfigService,
        terminusOptionsService: TerminusOptionsService
      ): Promise<TerminusModuleOptions> => {
        // Do some logging.. etc

        return terminusOptionsService.createTerminusOptions()
      }
    }),
```
Problem is that for the args i..e configService and terminusOptionsService I got the following error

```
Type '(configService: ConfigService, terminusOptionsService: TerminusOptionsService) => Promise<TerminusModuleOptions>' is not assignable to type '(...args: unknown[]) => TerminusModuleOptions | Promise<TerminusModuleOptions>'.
  Types of parameters 'configService' and 'args' are incompatible.
    Type 'unknown' is not assignable to type 'ConfigService'.
```

I am trying to use it like I use other modules but it doesn't seem to allow me to inject providers 

Any ideas ?

## Expected behavior
To be able to use forRootAsync and useFactory and inject other things into the useFactory signature.

## Minimal reproduction of the problem with instructions
->

## Environment

<pre><code>
Nest version: 6.1
 
For Tooling issues:
- Node version: 11
- Platform:  MAC
Others:

</code></pre>

**Comment by appsolutegeek** [comment]
P.s ConfigService is a globalProvider and I use it similar in graphql i.e.

```
    GraphQLModule.forRootAsync({
      inject: [ConfigService, LoggerFactoryService],
      useFactory: async (configService: ConfigService, loggerFactoryService: LoggerFactoryService): Promise<GqlModuleOptions> => {
     // Code here...
      }
    }),
```

**Comment by BrunnerLivio** [comment]
Thanks for submitting. I am not able to reproduce the error, however it may be fixed with 6.1.5.
Can you check again in 6.1.5 `npm i -s @nestjs/termins@6.1.5`?
